### PR TITLE
OCPBUGSM-29476 Trigger BMH's deprovision when the ISODownloadURL changes

### PR DESCRIPTION
**Pull request by flaper87**
There are cases when the InfraEnv or ClusterDeployment parameters are chagned, which requires AI
to rebuild the discovery ISO. When this happens, BMAC must make sure that the BMH is properly updated
and that care is taken so that the existing image can be properly deprovisioned before the new one is
added.

This PR does that by comparing the existing ISODownloadURL value with the one set in the BMH. When they difer,
BMAC assumes that a deprovision of the existing URL must happen and the new one should be, eventually, provisioned.

Note that the AI base url doesn't change and this PR relies on the signature of the URL to change. This will only
happen when AI is running with authentication enabled. In a non-auth environment, this workflow won't work.

Signed-off-by: Flavio Percoco <flavio@redhat.com>

### CODE REVIEW

File: internal/controller/controllers/bmh_agent_controller_test.go [hunk 1]
```diff
@@ -842,8 +845,12 @@ var _ = Describe("bmac reconcile", func() {
 			})
 		})
 
-		Context("when bmh.Spec.Image exists", func() {
+		Context("when bmh.Spec.Image doesn't need to be updated", func() {
 			It("the detached annotation is not removed", func() {
+				host.Status.Provisioning.State = bmh_v1alpha1.StateProvisioned
+				host.Status.OperationHistory.Provision.Start = metav1.Time{Time: time.Now()}
```
Comment (flaper87) [inline internal/controller/controllers/bmh_agent_controller_test.go:851]: We don't really need to set this, this was relevant for the original implementation. I will remove it if there are changes requested on this PR.

Comment (djzager) [reply internal/controller/controllers/bmh_agent_controller_test.go:851]: Now that you need to rebase anyway it's a perfect time to do it :sunglasses:

**Review by djzager — APPROVED** [review]

**Pull request merged**